File: Data Structures & Algorithms/longest-consecutive-sequence/submission-18.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>st(nums.begin(),nums.end());
        int ans=0;
        for(int i:nums)
        {
            if(!st.count(i-1))
            {
                int count=1;
                int x=i+1;
                while(st.count(x))
                {
                    count++;
                    x++;
                }
                ans=max(ans,count);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-18.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer>mp1=new HashMap<>();
        for(int i:nums)
        {
            mp1.put(i,mp1.getOrDefault(i,0)+1);
        }
        int[] ans=new int[k];
        PriorityQueue<Integer>pq=new PriorityQueue<>((a,b)->mp1.get(a)-mp1.get(b));
        for(int i:mp1.keySet())
        {
            pq.offer(i);
            if(pq.size()>k)
            {
                pq.poll();
            }
        }
        int i=0;
        while(!pq.isEmpty())
        {
            ans[i]=pq.peek();
            pq.poll();
            i++;
        }
        return ans;
    }
}
